### PR TITLE
Add documentation about TypeTransformers

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Adds documentation to `azure-arm.yaml` to explain how TypeTransformers work.

**Special notes for your reviewer**:

Including the documentation in this file ensures it is right to hand when needed.

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains documentation

### CODE REVIEW

File: v2/azure-arm.yaml [hunk 1]
```diff
@@ -130,8 +130,45 @@ typeFilters:
...
+# Selection Properties
+#
+# group    - select types or properties in the specified group or groups
+# version  - select types or properties in the specified version or version(s)
+# name     - select types or properties with the specified name or name(s)
+# property - select properties with the given name or name(s)
+#
+# All of these properties are matchers:
+#
+# o They wllow multiple values, semicolon (;) separated
```
Comment (super-harsh) [inline v2/azure-arm.yaml:153]:
```suggestion
# o They allow multiple values, semicolon (;) separated
```

Comment (super-harsh) [reply v2/azure-arm.yaml:153]: At first, I thought that `o` was a character there by mistake and not a bullet point

**Review by super-harsh — APPROVED** [review]

File: v2/azure-arm.yaml (hunk 1 above)
Comment (theunrepentantgeek) [reply v2/azure-arm.yaml]: Normally when I write bullet points like this, I indent like so:

``` go
#
# All of these properties are matchers:
#
#   o   They wllow multiple values, semicolon (;) separated
```

which makes it more obvious. However, our ASO linting objects, so this is my compromise.

**Pull request merged**